tests/components/history/test_init.py

```diff
@@ -188,6 +189,39 @@ def set_state(state):
 
         assert states == hist[entity_id]
 
+    def test_ensure_state_can_be_copied(self):
```

Comment:
Ha, I was just looking at the bug. Any reason why that integration even copies it?

Reply:
Its passing history states and core states though that function.  AFAICT, It needs to copy the core state to avoid mutating it.  It probably doesn't need to copy the ones it gets from history though.